docs/1-introduction/1-1-getting-started/1-1-3-hello-triangle.md

```diff
@@ -1,11 +1,456 @@
+    vertexShaderBlob->GetBufferPointer(),
+    vertexShaderBlob->GetBufferSize(),
+    &_vertexLayout)))
+{
+    std::cout << "D3D11: Failed to create default vertex input layout\n";
+    return false;
+}
+```
+
+As usual we follow the same pattern, we pass in our `vertexInputLayoutInfo` that we just created and its size, we also need to pass our vertex blob pointer and size, and finally our output parameter which is our input layout.
+
+Now all we have to do is create a vertex buffer (don't worry it's really easy) and issue our first `Draw` command!
+
+!!! error
+    Image showing stride and offset?
+
+## Vertex Buffers
+
+Vertex Buffers might seem hard at first, but they're really nothing more than a buffer that resides in our device's memory, which means really fast access. This buffer will be then bound and read by the vertex shader.
+
+Creating a vertex buffer is also really easy, first of all we have to make some data to put in our buffer, since we want to draw a triangle, we will be creating 3 vertices using our `VertexPositionColor` struct.
+
+```cpp
+constexpr VertexPositionColor vertices[] =
+{
+    { Position{  0.0f,  0.5f, 0.0f }, Color{ 0.25f, 0.39f, 0.19f } },
+    { Position{  0.5f, -0.5f, 0.0f }, Color{ 0.44f, 0.75f, 0.35f } },
+    { Position{ -0.5f, -0.5f, 0.0f }, Color{ 0.38f, 0.55f, 0.20f } },
+};
+```
+
+There we go, remember the `NDC` coordinates we talked about earlier? We are storing these coordinates that form a triangle here, we're also storing a color for each vertex, since they will be interpolated by our pixel shader.
+
+If you want you can try to visualize the triangle, take a piece of paper, draw a **Cartesian Plane**, draw 3 points and connect the dots with these coordinates.
+
+Now that we have the data, let's store this on our vertex buffer:
+
+```cpp
+D3D11_BUFFER_DESC bufferInfo = {};
+bufferInfo.ByteWidth = sizeof(vertices);
+bufferInfo.Usage = D3D11_USAGE::D3D11_USAGE_IMMUTABLE;
+bufferInfo.BindFlags = D3D11_BIND_FLAG::D3D11_BIND_VERTEX_BUFFER;
+D3D11_SUBRESOURCE_DATA resourceData = {};
+resourceData.pSysMem = vertices;
+if (FAILED(_device->CreateBuffer(
+    &bufferInfo,
+    &resourceData,
+    &_triangleVertices)))
+{
+    std::cout << "D3D11: Failed to create triangle vertex buffer\n";
+    return false;
+}
+```
+
+We begin by filling a `bufferInfo` descriptor for our buffer, we specify how many bytes we want, since this buffer will never change, for the `Usage` we specify: `D3D11_USAGE_IMMUTABLE`, this lets D3D11 put this data a close as possible to the GPU, finally we specify how we want to use this buffer, we want this to be a vertex buffer, so for the `BindFlags` we give: `D3D11_BIND_VERTEX_BUFFER`.
+
+And finally we create a `resourceData`, and populate the only field we care about: `pSysMem`, which is a pointer to our vertices which are currently in system RAM.
+
+Then we issue the creation of the buffer using `CreateBuffer`, using the informations we collected until now.
+
+## Drawing our triangle
```

Comment:
In this section, make a reference to the initial pipeline overview... since we call all the necessary stages in the same order.. IA/VS/PS/RS/OM :) should be some sort of epiphany moment for people here